budget_crm/test/test_crm_to_budget.yml

```diff
@@ -0,0 +1,110 @@
+-
+   I select for this file a user that is not admin
+-
+  !context
+    uid: 'res_users_colette'
+-
+  I first create a budget item
+-
+  !record {model: budget.item, id: budget_item_1}:
+    code: "BI1"
+    name: "Budget item 1"
+    account: [account.a_expense]
+-
+  I then create a budget
+-
+  !record {model: budget.budget, id: budget_test_1}:
+    name: "Budget test"
+    start_date: !eval "'%s-01-01' %(datetime.now().year)"
+    end_date: !eval "'%s-12-31' %(datetime.now().year)"
+    budget_item_id: budget_item_1
+-
+  and a budget version
+-
+  !record {model: budget.version, id: budget_version_1}:
+    name: "Budget Version 1"
+    budget_id: budget_test_1
+    currency_id: base.EUR
+-
+   Edit budget lines
+-
+  !record {model: budget.line, id: budget_line_1}:
+    name: first budget line description
+    analytic_account_id: account.analytic_agrolait
+    budget_item_id: budget_item_1
+    amount: 200.0
+    currency_id: base.EUR
+    budget_version_id: budget_version_1
+-
+  and a second budget line without description
+-
+  !record {model: budget.line, id: budget_line_2}:
+    analytic_account_id: account.analytic_agrolait
+    budget_item_id: budget_item_1
+    amount: 300.0
+    currency_id: base.EUR
+    budget_version_id: budget_version_1
+-
+   I select for this file a user that is not admin
+-
+  I make version active
+-
+  !python {model: budget.version}: |
+    model.make_active(cr, uid, [ref('budget_version_1')], context)
+-
+  I set the active budget version in the company
+-
+  !record {model: res.company, id: base.main_company}:
+    budget_version_id: budget_version_1
+-
+  I create a lead
+-
+  !record {model: crm.lead, id: lead_2}:
+    name: "Sell something"
+    date_deadline:  !eval "'%s-03-15' %(datetime.now().year)"
```

Comment:
I am unsure about this. While past dates like 2012 will always be past, here you create a budget on the current year. Could this test have different behavior before and after march 15th?